public/calendar_reboot.js

```diff
@@ -808,32 +826,63 @@ Calendar.prototype.fillCalendar = function(monthObjectsArray) {
 var LocalCalendarStorage = function(params) {
     var self = this;
     var prefix = params['storeId'] || "";
+    var event = params['event'];
     var allCalendarIdsKey = 'allCalendarIdsKey';
     //the current_active_calendar is the key for localStorage that stores
     //the active calendar's Id
     var current_active_calendar = 'current_active_calendar';
+    // This tells you whether the storage is actively working.
+    self.activeCalls = 0;
     
-    var toKey = function(id) {
+     var toKey = function(id) {
         //make a key out of a uniqueId
         
         var key = prefix + "_" + id;
         return key;
     };
     
+    self.startWork = function() {
+        // Will increment the counter and possibly fire an event.
+        console.log("Accessing storage");
+        self.activeCalls += 1;
+        
+        // Will dispatch the event backgroundActivityChange 
+        console.log("dipatching backgroundActivityChange");
+        window.dispatchEvent('backgroundActivityChange');
+    };
+    
+    self.endWork = function() {
+        // Will decrement the counter and maybe fire an event.
+        console.log("Done accessing storage");
+        self.activeCalls -= 1;
+        
+        // Will cancel the backgroundActivityChange event (stopPropagation)?
+      
+    };
+    
     self.getAllCalendarIds = function() {
         // Returns a promise for the allCalendarIds object from storage
         
         return new Promise( function(resolve, reject) {
+            self.startWork();
             var allCalendarIds = loadFromLocalStorage(toKey(allCalendarIdsKey));
             
             if (allCalendarIds !== null ) {
-                
                 resolve(allCalendarIds);
             }
             else {
                 reject("Not found");
             }
         })
+        .then( function() {
+            
+            self.endWork();
+        }.bind(this))
+        .catch( function() {
+            
+            self.endWork();
```

Comment:
You had to duplicate the call to `endWork()` in two places (in `.then()` and in `.catch()`) because you don't know which of the two will happen.

The error message is the one that is passed to this function as an argument. Whatever you called `reject` with.